src/assets/scripts/client/airport/MapCollection.js

```diff
@@ -0,0 +1,140 @@
+import _forEach from 'lodash/forEach';
+import _has from 'lodash/has';
+import _isArray from 'lodash/isArray';
+import _isObject from 'lodash/isObject';
+import BaseCollection from '../base/BaseCollection';
+import MapModel from './MapModel';
+
+/**
+ * Collection of `MapModel` objects
+ *
+ * Provides methods to create `MapModel` objects, used by `AirportModel`
+ * and to extract video map line data to be rendered by the
+ * `CanvasController`
+ *
+ * @class MapCollection
+ * @extends BaseCollection
+ */
+export default class MapCollection extends BaseCollection {
+    /**
+     * @constructor
+     * @param mapJson {object}
+     * @param airportPositionModel {StaticPositionModel}
+     * @param magneticNorth {number}
+     */
+    constructor(mapJson, airportPositionModel, magneticNorth) {
+        super();
+
+        if (!_isArray(mapJson) && !_isObject(mapJson)) {
+            throw new TypeError(
+                `Invalid parameter passed to MapCollection. Expected an array or object but found ${typeof mapJson}`
+            );
+        }
+
+        /**
+         * @inherited
+         * @memberof BaseCollection
+         * @property _items
+         * @type {array<MapModel>}
+         * @default []
+         */
+
+        /**
+         * @inherited
+         * @memberof BaseCollection
+         * @property length
+         * @type {number}
+         * @default #_items.length
+         */
+
+        /**
+         * @property _airportPositionModel
+         * @type {StaticPositionModel}
+         * @default null
+         * @private
+         */
+        this._airportPositionModel = null;
+
+        /**
+         * @property _magneticNorth
+         * @type {number}
+         * @default null
+         * @private
+         */
+        this._magneticNorth = null;
+
+        this._init(mapJson, airportPositionModel, magneticNorth);
+    }
+
+    /**
+     * Public fascade for `#_items`
+     *
+     * @property maps
+     * @return {array<MapModel>}
+     */
+    get maps() {
+        return this._items;
+    }
+
+    /**
+     * @for MapCollection
+     * @method getVisibleMapLines
+     * @return {array<array<object>>}
+     */
+    getVisibleMapLines() {
+        const filtered = this._items.filter((map) => {
```

Comment:
this can be simplified by using `reduce`:
```js
return this._items.reduce((sum, map) => {
    if (map.isHidden || map.lines.length === 0) {
        return sum;
    }

    return [
        ...map.lines,
        ...sum,
    ];
}, []);
```

Reply:
Clever.

Reply:
haha, perhaps.  It just means 1 loop instead of 2 😉